compiler/rustc_errors/src/lib.rs

```diff
@@ -556,7 +556,7 @@ impl Drop for HandlerInner {
         // instead of "require some error happened". Sadly that isn't ideal, as
         // lints can be `#[allow]`'d, potentially leading to this triggering.
         // Also, "good path" should be replaced with a better naming.
-        if !self.has_any_message() && !self.suppressed_expected_diag {
+        if !self.has_any_message() && !self.suppressed_expected_diag && !std::thread::panicking() {
```

Comment:
It might make sense to also add this at delayed_span_bugs? When there's a panic, I'm much more likely to be interested in the panic than the delayed bug. The delayed bug is actually not a bug if compilation stops with an error later -- they are used for cases where "if this happens we must never produce an output file, but we can't `bug!` here and now since compilation continues when an error was found, to report more than one error at a time". In those cases, when a panic occurred, all is good -- no output file is produced, after all.

Reply:
I'm gonna leave that up to another PR.

I actually slighlty disagree, or at least think it needs a bit more nuance than the delayed_good_path_bug case -- I think that *some* delayed span bugs are noise (e.g. dropping opaque type storage), but some others are actually pretty important in understanding why the compiler got so messed up that it's now panicking (e.g. diagnostics that are `.delay_as_bug()`'d).